tests/auth.test.ts

```diff
@@ -0,0 +1,178 @@
+import { AuthenticationTypes, SecurityDefinition, BasicSecurityDefinition } from '../src';
+import { getAuthentication } from '../src/swagger/auth';
+
+const BASIC_AUTH_ID = 'BasicAuth';
+const API_KEY_AUTH_ID_1 = 'APIKeyHeader';
+const API_KEY_AUTH_ID_2 = 'APIKeyHeader2';
+const API_KEY_HEADER_1 = 'X-API-KEY';
+const API_KEY_HEADER_2 = 'X-API-KEY-2';
+
+const basicSecurityDefinition: BasicSecurityDefinition = {
+  id: BASIC_AUTH_ID,
+  type: AuthenticationTypes.BASIC,
+};
+const apiKeySecurityDefinition1 = {
+  id: API_KEY_AUTH_ID_1,
+  type: AuthenticationTypes.API_KEY,
+  header: API_KEY_HEADER_1,
+};
+const apiKeySecurityDefinition2 = {
+  id: API_KEY_AUTH_ID_2,
+  type: AuthenticationTypes.API_KEY,
+  header: API_KEY_HEADER_2,
+};
+
+describe('getAuthentication', () => {
+  it('should return basic auth config', () => {
+    const securityDefinitions: SecurityDefinition[] = [
+      basicSecurityDefinition,
+    ]
+    const auth = {
+      securityDefinitions,
+      security: [[BASIC_AUTH_ID]],
+    };
+    const result = getAuthentication(auth);
```

Comment:
🤮 Had to explicitly define `SecurityDefinition []` here for `getAuthentication` to work without type errors 😞